### DISCUSSION TITLE
Open a stack screen with `useLinking` without leaving your current tab

### DISCUSSION THREAD

**Discussion by nandorojo** [react-navigation/react-navigation#10086] [Ideas]
**TLDR**: Every tab is rendering the same stack, and when I navigate to a URL, I want the screen to open on the focused tab (if possible). This is similar to how Twitter lets you open a Tweet from both the feed _or_ the profile.

I am trying to do all linking with the `useLinking` hook. I really like this API, since I only trigger via URLs, which ports well to web.

However, I am having an issue. I want to be able to open a given screen inside of my _current_ tab. 

Here's my architecture:

`RootTabs`
  - `SearchTab`
  - `DiscoverTab`

Both `SearchTab` and `DiscoverTab` tabs have the same `RootStack`:

`RootStack`
 - `Search`
 - `Discover`
 - `Artist`

Each tab uses the same `RootStack`; the only difference is that they pass a different `initialRouteName` to the stack.

This all works well with my linking config.

However, when I call `linkTo('/artist/123')` from the `SearchTab`, it goes to the `DiscoverTab` and opens the artist there.

This doesn't match the pattern I want.

Instead, I want to be able to stay on the `SearchTab`, and open the `Artist` screen on top of it.

Likewise, if I'm on the `DiscoverTab`, and I push an `Artist`, I want it to open on top of `Discover`.

How should I do this? Does this require a custom `getStateFromPath`?
